include/rdma: New options to fi_getopts for EFA

lrbison opened the pull request:
* Added three new options for fi_getopts in fi_ext.h: FI_OPT_EFA_EMULATED_READ FI_OPT_EFA_EMULATED_WRITE FI_OPT_EFA_EMULATED_ATOMICS

 * Added documentation in fi_efa.7.md These options can be used by libfabric users to query if an EFA endpoint is using emulation or has hardware support for the Read, Write, and Atomics operations.

j-xiong commented:
Provider specific options have their own range and should not be here. See `fi_ext.h` for examples of how provider specific options are defined.

shefty reviewed (changes requested):
Provider defines go into fi_ext.h.

lrbison commented:
Thank you @shefty and @j-xiong, I'm sorry for not realizing that.

I've moved the documentation to fi_efa's .md file and the definitions to fi_ext.h, and updated the commit message.

wenduwan commented on man/fi_efa.7.md:
```diff
@@ -97,6 +97,11 @@ provider for AWS Neuron or Habana SynapseAI.
 
+*FI_OPT_EFA_EMULATED_READ, FI_OPT_EFA_EMULATED_WRITE, FI_OPT_EFA_EMULATED_ATOMICS - bool*
+: These options only apply to the fi_getopt() call.
+  They are used to query the EFA provider to determine if the endpoint is  emulating Read, Write, and Atomic operations (return value is true), or if
```
Telling from the rest of the file, this line has probably exceeded the max length.

lrbison replied:
Looks like a missed line ending and didn't notice because I keep my window narrow.  Thank you, fixed!

lrbison commented:
@shefty gentle ping on this fi_getopts change, is the new location in fi_ext.h appropriate?

shefty commented:
Yes, this is good -- thanks!
